app/services/solidus_square/payments/create.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  module Payments
+    class Create < ::SolidusSquare::Base
+      attr_reader :client, :source_id, :amount, :auto_complete
+
+      def initialize(client:, source_id:, amount:, auto_complete:)
+        @client = client
+        @source_id = source_id
+        @amount = amount
+        @auto_complete = auto_complete
+
+        super
+      end
+
+      def call
+        create_payment
+      end
+
+      private
+
+      def create_payment
+        handle_square_result(client.payments.create_payment(body: payment_payload)) do |result|
+          result.body&.payment
+        end
+      end
+
+      def payment_payload
+        {
+          idempotency_key: idempotency_key,
+          source_id: source_id,
+          amount_money: {
+            amount: amount,
+            currency: "USD"
+          },
+          autocomplete: auto_complete
```

Comment:
autocomplete on square should be always false.

This is because even If `Spree::PaymentMethod#auto_capture?` is set to true, we'll capture the payment only when the order is completed.